software/runtime/runtime.h

```diff
@@ -81,8 +98,68 @@ static inline void mempool_wfi() { asm volatile("wfi"); }
 
 // Wake up core with given core_id by writing in the wake up control register.
 // If core_id equals -1, wake up all cores.
-static inline void wake_up(uint32_t core_id) { wake_up_reg = core_id; }
+static inline void wake_up(uint32_t volatile core_id) { wake_up_reg = core_id; }
```

Comment:
The `volatile` in the function argument should not have any impact. Did you have problems with the wake_up_reg being written?

Reply:
I think this is just a leftover of debugging the partial barriers. The barriers work without it.